test/controllers/income/UpdateIncomeNextYearControllerSpec.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.income
+
+import builders.{AuthBuilder, RequestBuilder}
+import controllers.{ControllerViewTestHelper, FakeTaiPlayApplication}
+import mocks.{MockPartialRetriever, MockTemplateRenderer}
+import org.mockito.Matchers
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mock.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import play.api.i18n.Messages.Implicits._
+import play.api.mvc.Result
+import play.api.test.Helpers._
+import uk.gov.hmrc.domain.Generator
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.frontend.auth.connectors.domain.Authority
+import uk.gov.hmrc.play.frontend.auth.connectors.{AuthConnector, DelegationConnector}
+import uk.gov.hmrc.play.partials.FormPartialRetriever
+import uk.gov.hmrc.renderer.TemplateRenderer
+import uk.gov.hmrc.tai.model.cache.UpdateNextYearsIncomeCacheModel
+import uk.gov.hmrc.tai.service.{PersonService, UpdateNextYearsIncomeService}
+import views.html.incomes.nextYear.updateIncomeCYPlus1Start
+
+import scala.concurrent.Future
+import scala.util.Random
+
+class UpdateIncomeNextYearControllerSpec extends PlaySpec
+  with FakeTaiPlayApplication
+  with MockitoSugar
+  with ControllerViewTestHelper {
+
+  "start" must {
```

Comment:
I think we're missing a test case of when the service doesn't return details (currently an error but this can be improved) and when the route is called and the feature is toggled off (since this is not dynamic and is static config we might need to mock the config 😞 )